src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -91,7 +91,14 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     $emptyData = $form->getConfig()->getEmptyData();
 
                     if (false === FormUtil::isEmpty($emptyData) && array() !== $emptyData) {
-                        $data = is_callable($emptyData) ? call_user_func($emptyData, $form, $data) : $emptyData;
+                        if (is_callable($emptyData)) {
+                            if (is_string($emptyData)) {
```

Comment:
I've introduced that change as a bug fix in #17822 targeting 2.7.
But I've made a mistake, IMO we should do it like https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Form.php#L610 and merge it in 2.7 as bug fix too. No need to deprecate anything.

Reply:
#17822 is over 1.5y old now. People might have built code against that "feature" already. :-(

Reply:
Also, I would not check for `instanceof \Closure` as this would forbid array callables and invokable objects for no reason.

Reply:
The `empty_data` MUST be a closure for all forms, I'm not sure people rely on this hidden feature which is just a bug I introduced no matter when IMO.